### ISSUE TITLE
Payment status stuck at pending

### ISSUE THREAD

**Issue by dpostma** [Cloudstek/whmcs-mollie-payment-gateway#3]
No idea what to write here...

**Steps to reproduce:**

I don't know, sorry.

**WHMCS version:** 7.1.1
**PHP version:** 5.6
**Mollie payment gateway version:** Newest
**Theme:** https://themeforest.net/item/spark-responsive-hosting-technology-wordpress-theme/18146709

**Enabled addons:**

Proxmox module
Lara template settings

**Enabled payment gateways:**

PayPal
iDeal
Overboeking

**Relevant gateway log output:**

Payment attempted for invoice 178. Awaiting payment confirmation from callback for transaction tr_SbxNN2svTQ.

**Additional information:**

* Problem can be reliably reproduced, doesn't happen randomly: [No]
* Problem started happening recently: [No]
* Problem can be reproduced in an older version: [no idea]

Basically the status will not update to active but is stuck on pending instead.

**Comment by mdeboer** [comment]
Thanks for reporting, based on other emails I received I think it has to do with a WHMCS update. Currently have my hands full on some projects but I will try to fix this in my hours off.

**Comment by DeSavanger** [comment]
Same problem here.

**Comment by dpostma** [comment]
UPDATE: It does work, you have to change your WHMCS URL to http**s** in case you're using SSL.

**Comment by mdeboer** [comment]
@WhiteSasch can you try the solution @dpostma has provided in case you're using SSL? If this fixes the problem for you I'll close this issue but for now I'll leave it open.

**Comment by DeSavanger** [comment]
@mdeboer I can't do that, because I'm using CloudFlare to auto use SSL, so when I change the WHMCS URL in my WHMCS config to https://, it will give a redirection error in the browser...

**Comment by DeSavanger** [comment]
@mdeboer https://gyazo.com/9d311032736f247cb6f70ffa37b43a02

**Comment by dpostma** [comment]
I also use CloudFlare for SSL. Everything works just fine for me.

EDIT: I have a few page rules added in my CF config which redirects all HTTP trafic to HTTPS. Try to configure that.

**Comment by DeSavanger** [comment]
@dpostma I had also that page rule which redirects all traffic from HTTP to HTTPS. I've now removed that page rule, set in DirectAdmin "Secure SSL" to "true" and "Create a symbolic link from private_html to public_html". I've set my CloudFlare SSL mode to "Full" instead of "Flexible". Now I've changed my WHMCS system URL in the WHMCS config and everything works fine! Issue resolved, no redirect errors anymore and Mollie works fine! My invoice shows "Paid".
